app/scripts/modules/amazon/src/loadBalancer/configure/common/LoadBalancerLocation.tsx

```diff
@@ -111,51 +115,102 @@ class LoadBalancerLocationImpl extends React.Component<ILoadBalancerLocationProp
   }
 
   public componentDidMount(): void {
-    if (this.props.isNew || this.props.forPipelineConfig) {
-      this.loadAccounts();
-      const hideInternalFlag = this.shouldHideInternalFlag();
-      this.setState({ hideInternalFlag });
-    }
-
+    this.setState({ hideInternalFlag: this.shouldHideInternalFlag() });
     if (this.props.loadBalancer && this.props.isNew) {
       this.buildName();
     }
-  }
 
-  private loadAccounts(): void {
-    AccountService.listAccounts('aws').then(accounts => {
-      this.setState({ accounts });
-      this.accountUpdated(this.props.formik.values.credentials);
+    const formValues$ = this.props$.map(props => props.formik.values);
+    const selectedAccount$ = formValues$.map(x => x.credentials).distinctUntilChanged();
+    const selectedRegion$ = formValues$.map(x => x.region).distinctUntilChanged();
+    const subnetPurpose$ = formValues$.map(x => x.subnetType).distinctUntilChanged();
+
+    const appName$ = this.props$.map(props => props.app.name).distinctUntilChanged();
+    const stack$ = formValues$.map(x => x.stack).distinctUntilChanged();
+    const detail$ = formValues$.map(x => x.detail).distinctUntilChanged();
+
+    const accounts$ = Observable.fromPromise(AccountService.listAccounts('aws')).shareReplay(1);
+
+    const regions$ = Observable.combineLatest(selectedAccount$, accounts$) // wait for accounts to load and be cached
+      .switchMap(([selectedAccount, _accounts]) => AccountService.getRegionsForAccount(selectedAccount))
+      .shareReplay(1);
+
+    const allLoadBalancers$ = this.props.app.getDataSource('loadBalancers').data$ as Observable<IAmazonLoadBalancer[]>;
+    const existingLoadBalancerNames$ = Observable.combineLatest(allLoadBalancers$, selectedAccount$, selectedRegion$)
+      .map(([allLoadBalancers, selectedAccount, selectedRegion]) => {
+        return allLoadBalancers
+          .filter(lb => lb.account === selectedAccount && lb.region === selectedRegion)
+          .map(lb => lb.name);
+      })
+      .shareReplay(1);
+
+    const subnets$ = Observable.combineLatest(selectedAccount$, selectedRegion$)
```

Comment:
This is really minor but it's easy to confuse `subnets$` and `subnet$` when reading the code; maybe rename this to `availableSubnets$` or `allSubnets$`?

Reply:
I made some changes to the variable names in a new commit to address this, see below